packages/cli/lib/marketplace-validate.js

```diff
@@ -0,0 +1,138 @@
+const chalk = require('chalk');
+
+const { logger } = require('@hubspot/cli-lib/logger');
+const {
+  requestValidation,
+  getValidationStatus,
+  getValidationResults,
+} = require('@hubspot/cli-lib/api/marketplaceValidation');
+const { i18n } = require('@hubspot/cli-lib/lib/lang');
+const { EXIT_CODES } = require('./enums/exitCodes');
+
+const SLEEP_TIME = 2000;
+
+const kickOffValidation = async (accountId, assetType, src) => {
+  const requestGroup = 'EXTERNAL_DEVELOPER';
+  try {
+    const requestResult = await requestValidation(accountId, {
+      path: src,
+      assetType,
+      requestGroup,
+    });
+    return requestResult;
+  } catch (err) {
+    logger.debug(err);
+    process.exit(EXIT_CODES.ERROR);
+  }
+};
+
+const pollForValidationFinish = async (accountId, validationId) => {
+  try {
+    const checkValidationStatus = async () => {
+      const validationStatus = await getValidationStatus(accountId, {
+        validationId,
+      });
+
+      if (validationStatus === 'REQUESTED') {
+        await new Promise(resolve => setTimeout(resolve, SLEEP_TIME));
```

Comment:
@ssethia2 is the Kafka producer ready for this yet? We can switch to that when it is

Reply:
We have that planned as a low priority issue, but thanks for calling that out, I'll update this when we eventually do move to using that